### PR TITLE
Update code quality for Assignment 1

### PR DESCRIPTION
**Pull request by JrmCkh**
Updated as per comments

### CODE REVIEW

File: backend/server/services/questionService.js [hunk 1]
```diff
@@ -1,24 +1,20 @@
 const questionRepository = require('../repositories/questionRepository');
 
-function missingInputsThrowsValidationError(title, complexity, description) {
+const missingInputsThrowsValidationError = (title, complexity, description) => {
   const innerText = description.replace(/<[^>]+>|\s+/g, '');
   if (!title || !complexity || !innerText) {
     throw { status: 400, message: 'Missing inputs' };
   }
 }
```
Comment (anthonynmh) [inline backend/server/services/questionService.js:8]: Should have a semicolon here

File: backend/server/services/questionService.js [hunk 2]
```diff
@@ -1,24 +1,20 @@
...
 
-    const isCurrent = await questionRepository.findByIdAndTitle(id, title);
-    if (!isCurrent) {
-      throw { status: 409, message: `Question title already exist` };
-    }
+  // Duplicate found, check if it is the same question
+  if (!isDuplicateTitle._id.equals(id)) {
+    throw { status: 409, message: `Question title already exist` };
   }
 }
```
Comment (anthonynmh) [inline backend/server/services/questionService.js:19]: Same as above

**Review by anthonynmh — CHANGES_REQUESTED** [review]
Minor standardisation

File: backend/server/services/questionService.js (hunk 1 above)
Comment (JrmCkh) [reply backend/server/services/questionService.js]: Resolved

File: backend/server/services/questionService.js (hunk 2 above)
Comment (JrmCkh) [reply backend/server/services/questionService.js]: Resolved

**Review by cadencjk — APPROVED** [review]
LGTM!

**Review by anthonynmh — APPROVED** [review]
LGTM

**Pull request merged**